pkg/execution/executor/executor.go

```diff
@@ -80,6 +92,29 @@ type Executor interface {
 		// right order.
 		stackIndex int,
 	) (*state.DriverResponse, int, error)
+
+	// HandlePauses handles pauses loaded from an incoming event.  This delegates to Cancel and
+	// Resume where necessary, depending on pauses that have been loaded and matched.
+	HandlePauses(ctx context.Context, iter state.PauseIterator, event event.TrackedEvent) error
+	// Cancel cancels an in-progress function, preventing any enqueued or future steps from running.
+	Cancel(ctx context.Context, id state.Identifier, r CancelRequest) error
+	// Resume resumes an in-progress function from the given waitForEvent pause.
+	Resume(ctx context.Context, p state.Pause, r ResumeRequest) error
+	// SetFailureHandler sets the failure handler, called when a function permanently fails.
+	SetFailureHandler(f FailureHandler)
+}
+
+// CancelRequest stores information about the incoming cancellation request within
+// history.
+type CancelRequest struct {
+	EventID    *ulid.ULID
+	Expression *string
+	UserID     *uuid.UUID
+}
+
+type ResumeRequest struct {
+	With    any
```

Comment:
What does `With` do?

Reply:
Seems like just assigning some arbitrary data.
https://github.com/inngest/inngest/pull/570/files#diff-ccd0679a9f24c84fef4f5d4d5fe2c32651d23886d85af8448bb968921c7b8a95R605

Reply:
Yeah.  This assigns arbitrary data to the step, allowing us to use that data in the SDK's function.